Report.cs

```diff
@@ -1,16 +1,80 @@
 ﻿using System;
+using System.IO;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using CMP1903M_Assessment_1.Debugging;
 
 namespace CMP1903M_Assessment_1
 {
+    /// <summary>
+    /// Handles reporting/presenting the resulting data to the user
+    /// </summary>
     class Report
     {
-        //Handles the reporting of the analysis
-        //Maybe have different methods for different formats of output?
-        //eg.   public void outputConsole(List<int>)
+        public Report(Analyse analysis)
+        {
+            Analysis = analysis;
+            reportText = GenerateReportText();
+        }
+        public Analyse Analysis { get; private set; }
+        string reportText = "";

```

Comment:
C# standard recommends private variables begin an underscore.
```suggestion
        string _reportText = string.Empty;
```